src/box/index_def.c

```diff
@@ -120,10 +120,6 @@ index_def_new(uint32_t space_id, uint32_t iid, const char *name,
 		if (opts->is_unique)
 			def->cmp_def->unique_part_count =
 				def->key_def->part_count;
-		if (def->cmp_def == NULL) {
-			index_def_delete(def);
-			return NULL;
```

Comment:
The `identifier_check` call above can't fail either because the identifier must have already been checked in `index_def_new_from_tuple()`. Let's also drop that check, update the comment to `index_def_new()` to state that it never fails, and drop `NULL` checks from all call sites while we are at it.